install/cupy_builder/_compiler.py

```diff
@@ -94,19 +94,27 @@ def _nvcc_gencode_options(cuda_version: int) -> list[str]:
                          ('compute_86', 'sm_86'),
                          ('compute_89', 'sm_89'),
                          ('compute_90', 'sm_90'),]
-            if cuda_version >= 12080:
+            if cuda_version < 12080:
+                arch_list.append('compute_90')
+            elif 12080 <= cuda_version < 12090:
                 arch_list += [('compute_100', 'sm_100'),
                               ('compute_120', 'sm_120'),
                               'compute_100']
-            else:
-                arch_list.append('compute_90')
+            elif 12090 <= cuda_version:
+                arch_list += [('compute_100f', 'sm_100'),
+                              ('compute_103f', 'sm_103'),
+                              ('compute_120f', 'sm_120'),
+                              ('compute_121f', 'sm_121'),
+                              'compute_100']
 
             if aarch64:
                 # JetPack 5 (CUDA 12.0-12.2) or JetPack 6 (CUDA 12.2+)
                 arch_list += [
                     ('compute_72', 'sm_72'),  # Jetson (Xavier)
                     ('compute_87', 'sm_87'),  # Jetson (Orin)
                 ]
+                if 12080 <= cuda_version:
+                    arch_list.append(('compute_101', 'sm_101'))
```

Comment:
Removing it for now as it seems JetPack 7.0 w/ Thor support comes with CUDA 13.0: https://developer.nvidia.com/embedded/develop/roadmap

```suggestion
```